### PR TITLE
Fixing problem with TachoCount

### PR DESCRIPTION
**Pull request by jabrena**
Fixed and tested on an EV3 Brick with a motor.
The solution is fine for me.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,7 +14,7 @@ plugins {
     id "com.github.johnrengelman.shadow" version "4.0.3"
 }
 
-version = '2.6.0-SNAPSHOT'
+version = '2.7.0-SNAPSHOT'
```
Comment (JakubVanek) [inline build.gradle]: Hi,

Why is this a 2.6.x to 2.7.x update?

According to https://semver.org/:

> Given a version number MAJOR.MINOR.PATCH, increment the:
>
> 1. MAJOR version when you make incompatible API changes,
> 2. MINOR version when you add functionality in a backwards compatible manner, and
> 3. PATCH version when you make backwards compatible bug fixes.
>
>Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

Comment (JakubVanek) [reply build.gradle]: However I haven't followed semver in this project closely as well; it just surprised me to see this version jump.

**Comment by JakubVanek** [comment]
Example of the broken diff:
broken

**Comment by jabrena** [comment]
You are right, I will upload again the change, meanwhile, I will refactor the example.

**Comment by JakubVanek** [comment]
I have fixed the diff, refactored the code and changed the version to `2.6.1-SNAPSHOT`.

However I think I found an issue with this polarity approach. Will calling `backward()` or `forward()` affect the behaviour of the `rotate()` function? If yes, then this solution is not working fully (because the existing code has never worked fully). I think that calling `forward()` or `backward()` on leJOS has no lasting effects on the state. See this code:
```java
package example;

import ev3dev.actuators.lego.motors.EV3LargeRegulatedMotor;
import lejos.hardware.port.MotorPort;
import lejos.robotics.RegulatedMotor;
import lejos.utility.Delay;
import lombok.extern.slf4j.Slf4j;

@Slf4j
class FwdBwdRotateTest {
    public static void main(String[] args) {
        LOGGER.info("Initializing motor class");
        RegulatedMotor motor = new EV3LargeRegulatedMotor(MotorPort.A);

        LOGGER.info("Set motor speed");
        motor.setSpeed(360);

        LOGGER.info("Move motor forward for 1 second");
        motor.forward();
        Delay.msDelay(1000);

        LOGGER.info("Stop the motor and wait 0,5 seconds");
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Calling rotate(360) - should move forward, i.e. in the same direction as the last move");
        motor.rotate(360, false);

        LOGGER.info("Move motor backward for 1 second");
        motor.backward();
        Delay.msDelay(1000);

        LOGGER.info("Stop the motor and wait 0,5 seconds");
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Calling rotate(360) - should move forward, i.e. in the oppposite direction to the last move");
        motor.rotate(360, false);
    }
}
```
Behaviour that I would expect on leJOS:
1. Motor starts rotating in direction A and then stops.
2. Motor rotates 360 degrees in the direction A and then stops.
3. Motor starts rotating in direction B (opposite to A) and then stops.
4. Motor rotates 360 degrees in the direction A and then stops.

Behaviour that I suspect would happen on ev3dev-lang-java:
1. Motor starts rotating in direction A and then stops.
2. Motor rotates 360 degrees in the direction A and then stops.
3. Motor starts rotating in direction B (opposite to A) and then stops.
4. Motor rotates 360 degrees in the **direction B** and then stops.

I think this will happen because kernel will take the motor as `inversed`, and as such it will also invert the `run-to-rel-pos` argument `position_sp`.

**Comment by JakubVanek** [comment]
I have also found this:
https://www.ev3dev.org/docs/tutorials/tacho-motors/#run-forever

> … now the motor is running. Let’s make the motor turn in the opposite direction, but a little slower. Using a negative value changes the direction…
> ```sh
> $ echo -300 > $MC/speed_sp
> $ echo run-forever > $MC/command
> ```

**Comment by JakubVanek** [comment]
`rotateTo()` could also be problematic.
```java
package example;

import ev3dev.actuators.lego.motors.EV3LargeRegulatedMotor;
import lejos.hardware.port.MotorPort;
import lejos.robotics.RegulatedMotor;
import lejos.utility.Delay;
import lombok.extern.slf4j.Slf4j;

@Slf4j
class FwdBwdRotateToTest {
    public static void main(String[] args) {
        LOGGER.info("Initializing motor class");
        RegulatedMotor motor = new EV3LargeRegulatedMotor(MotorPort.A);

        LOGGER.info("Set motor speed");
        motor.setSpeed(360);

        LOGGER.info("Move motor forward for 1 second");
        motor.forward();
        Delay.msDelay(1000);

        LOGGER.info("Stop the motor and wait 0,5 seconds");
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Calling rotateTo(360)");
        motor.rotateTo(360, false);

        LOGGER.info("Move motor backward for 0,1 second");
        motor.backward();
        Delay.msDelay(100);

        LOGGER.info("Stop the motor and wait 0,5 seconds");
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Calling rotateTo(360) - motor should move forward a little bit");
        motor.rotateTo(360, false);
    }
}
```
Behaviour that I would expect on leJOS:
1. Motor rotates in direction A for a second and then stops.
2. Motor rotates to the position `origin+360°`.
3. Motor rotates in direction B (opposite to A) brieflt and then stops.
4. Motor rotates back to the position `origin+360°` + it should go into direciton A.

**Comment by jabrena** [comment]
Good analysis Jakub, I will test with the examples today

**Comment by JakubVanek** [comment]
Test 1 (FwdBwdRotateTest) failed. Motor has rotated in the B direction, so usage of `forward()` / `backward()` changes the behavior of `rotate()`, which it shouldn't. Forward/backward should not invert the motor, they should only start its rotation in that direction once.

**Comment by JakubVanek** [comment]
Test 2 (FwdBwdRotateToTest) failed. Motor has moved in the B direction, so it probably aimed at `origin-360°`. This means that `forward()`/`backward()` affect the behavior of `rotateTo()`, which they shouldn't.

Updated source code:
```java
package example;

import ev3dev.actuators.lego.motors.EV3LargeRegulatedMotor;
import lejos.hardware.port.MotorPort;
import lejos.robotics.RegulatedMotor;
import lejos.utility.Delay;
import lombok.extern.slf4j.Slf4j;

@Slf4j
class FwdBwdRotateToTest {
    public static void main(String[] args) {
        LOGGER.info("Initializing motor class");
        RegulatedMotor motor = new EV3LargeRegulatedMotor(MotorPort.A);

        LOGGER.info("Set motor speed");
        motor.hold();
        motor.setSpeed(360);

        LOGGER.info("Move motor forward for 1 second");
        motor.forward();
        Delay.msDelay(1000);
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Calling rotateTo(360)");
        motor.rotateTo(360, false);
        Delay.msDelay(500);
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Move motor backward for 0,5 second");
        motor.backward();
        Delay.msDelay(500);
        motor.stop();
        Delay.msDelay(500);

        LOGGER.info("Calling rotateTo(360) - motor should move forward a little bit");
        motor.rotateTo(360, false);
        Delay.msDelay(500);
        motor.stop();
    }
}
```

**Comment by JakubVanek** [comment]
Alternative solution: https://github.com/ev3dev-lang-java/ev3dev-lang-java/pull/689

**Pull request closed without merging**